src/components/IButtonGroup/IButtonGroup.vue

```diff
@@ -52,24 +43,26 @@ const props = defineProps({
      */
     size: {
         type: String,
-        default: ''
+        default: undefined
     }
 });
 
-const size = useSize({ componentName, currentSize: props.size || buttonGroup.size });
+const componentSize = useComponentSize({ componentName, currentSize: props.size });
+const { disabled, size } = useInputState({
+    disabled: props.disabled,
+    size: componentSize.value
+});
 
-const isDisabled = computed(
-    () => props.disabled || buttonGroup.disabled || form.disabled || formGroup.disabled
-);
+const isDisabled = computed(() => {
+    return props.disabled || disabled.value;
```

Comment:
Don't forget to take `buttonGroup.disabled` into account here for nested groups.